shards/shards_test.go

```diff
@@ -244,3 +287,93 @@ func searcherForTest(t *testing.T, b *zoekt.IndexBuilder) zoekt.Searcher {
 
 	return searcher
 }
+
+func reposForTest(n int) (result []*zoekt.Repository) {
+	for i := 0; i < n; i++ {
+		result = append(result, &zoekt.Repository{
+			Name: fmt.Sprintf("test-repository-%d", i),
+		})
+	}
+	return result
+}
+
+func testSearcherForRepo(b testing.TB, r *zoekt.Repository, numFiles int) zoekt.Searcher {
+	builder := testIndexBuilder(b, r)
+
+	builder.Add(zoekt.Document{
+		Name:    fmt.Sprintf("%s/filename-%d.go", r.Name, 0),
+		Content: []byte("needle needle needle haystack"),
+	})
+
+	for i := 1; i < numFiles; i++ {
```

Comment:
Should this be `<= numFiles`?

Reply:
It can be, but I specifically wanted to add 1 "needle" file and N-1 "haystack" files.